GoogleDataTransport/GDTCORLibrary/GDTCORConsoleLogger.m

```diff
@@ -16,30 +16,34 @@
 
 #import "GDTCORLibrary/Public/GDTCORConsoleLogger.h"
 
+volatile NSInteger GDTCORConsoleLoggerLoggingLevel = GDTCORLoggingLevelErrors;
```

Comment:
Does this really need to be volatile? If so, please add an explanatory comment in the header.

Reply:
Added comment in the header, it's to ensure atomicity since this is a global variable that capable of being set from anywhere. I'll push the commit and merge after the checks pass.

Reply:
This isn't what `volatile` actually does in C. It's a common misconception because Java reused `volatile` to mean this.

In short, `volatile` instructs _the compiler_ that the variable may be modified through means it can't detect, forcing it to load and store from memory each time, but it does not force a memory barrier, so the CPU is still free reorder reads and writes to this memory.

In C, `volatile` is useful for memory mapped I/O and a few other niche uses, but does not guarantee atomicity, so it's almost certainly never useful to us in Firebase.

What you want is an `atomic_int` or you want just ignore the the threading issues and mark any comparison as exempt from TSAN checks.

Reply:
Makes sense. I'll add it to my list of TODOs to remove volatiles from GDT.